### PR TITLE
Add tf2 package

### PR DESCRIPTION
**Pull request by dbruggner**
This adds part of the geometry2 package (tf2).

File `repositories/console_bridge.BUILD.bazel` is stolen from https://github.com/mvukov/rules_ros2/compare/feature/pluginlib#diff-d1a7e3a17144dfb0f628076fc98f452c60b4d182f370b2f962ed23d518911ea1

### CODE REVIEW

File: repositories/deps.bzl [hunk 1]
```diff
@@ -4,6 +4,7 @@
 load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
 load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
 load("@rules_python//python:pip.bzl", "package_annotation")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
```
Comment (mvukov) [inline repositories/deps.bzl:7]: This can be removed.

File: repositories/patches/geometry2_remove_rclcpp_components.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+--- a/tf2_ros/src/static_transform_broadcaster_node.cpp
++++ b/tf2_ros/src/static_transform_broadcaster_node.cpp
+@@ -98,7 +98,3 @@ StaticTransformBroadcasterNode::StaticTransformBroadcasterNode(const rclcpp::Nod
+   broadcaster_->sendTransform(tf_msg);
+ }
+ }  // namespace tf2_ros
+-
+-#include "rclcpp_components/register_node_macro.hpp"
+-
+-RCLCPP_COMPONENTS_REGISTER_NODE(tf2_ros::StaticTransformBroadcasterNode)
```
Comment (mvukov) [inline repositories/patches/geometry2_remove_rclcpp_components.patch:12]: Just for my understanding, why is this needed?

**Review by mvukov — COMMENTED** [review]
Thanks for the contribution :+1:

**Comment by mvukov** [comment]
btw, do you use pre-commit hooks locally? I think the buildifier should've reformatted the build file.

**Comment by mvukov** [comment]
Something fishy is happening on CI. Please install pre-commit, install the hooks and then run `pre-commit run --all-files`. Then please commit and push the changes.

File: repositories/rclcpp.BUILD.bazel [hunk 3]
```diff
@@ -125,3 +125,15 @@ ros2_cpp_library(
         "@ros2_rcl_interfaces//:cpp_lifecycle_msgs",
     ],
 )
+
+ros2_cpp_library(
+    name = "rclcpp_components",
+    srcs = glob(["rclcpp_components/src/*.cpp"]),
+    hdrs = glob(["rclcpp_components/include/**/*.hpp"]),
+    includes = ["rclcpp_components/include"],
```
Comment (mvukov) [inline repositories/rclcpp.BUILD.bazel:133]: Per https://github.com/ros2/rclcpp/blob/16.0.1/rclcpp_components/CMakeLists.txt, I think we need finer specs for this library. https://github.com/ros2/rclcpp/blob/16.0.1/rclcpp_components/CMakeLists.txt#L24 suggested we may only need headers. Can you take a look at this?

File: repositories/geometry2.BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ros2_cpp_library(
+    name = "tf2_ros",
+    srcs = glob([
+        "tf2_ros/src/*.cpp",
+        "tf2_ros/src/*.hpp",
+    ]),
+    hdrs = glob([
+        "tf2_ros/include/**/*.h",
+        "tf2_ros/include/**/*.hpp",
+    ]),
```
Comment (mvukov) [inline repositories/geometry2.BUILD.bazel:55]: IMHO this also needs fine granularity: static_transform_broadcaster_node hdrs and srcs should go to a separate library for the moment like in CMake files. Do you need dynamic composition at the moment? At the moment this plugin-like system is not supported at the moment in this repo but we can add it if needed.

File: repositories/patches/geometry2_remove_rclcpp_components.patch (hunk 2 above)
Comment (dbruggner) [reply repositories/patches/geometry2_remove_rclcpp_components.patch:12]: `register_node_macro.hpp` is from `rclcpp_components` which depends on `ament_index` which isn't needed for bazel rules. `ament_index` is also removed in `rclcpp`.

File: repositories/geometry2.BUILD.bazel (hunk 4 above)
Comment (dbruggner) [reply repositories/geometry2.BUILD.bazel]: I removed the nodes, only the tf2_ros library is left.

File: repositories/rcl_interfaces.BUILD.bazel [hunk 5]
```diff
@@ -140,6 +140,12 @@ py_ros2_interface_library(
     deps = [":composition_interfaces"],
 )
 
+cpp_ros2_interface_library(
```
Comment (mvukov) [inline repositories/rcl_interfaces.BUILD.bazel:143]: This one can be removed as well.

File: repositories/geometry2.BUILD.bazel [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        "tf2_ros/src/buffer_server.cpp",
+        "tf2_ros/src/create_timer_ros.cpp",
+        "tf2_ros/src/static_transform_broadcaster.cpp",
+        "tf2_ros/src/transform_broadcaster.cpp",
+        "tf2_ros/src/transform_listener.cpp",
+    ],
+    hdrs = glob([
+        "tf2_ros/include/**/*.h",
+        "tf2_ros/include/**/*.hpp",
+    ]),
```
Comment (mvukov) [inline repositories/geometry2.BUILD.bazel:65]: Please exclude [static_transform_broadcaster_node.hpp](https://github.com/ros2/geometry2/blob/rolling/tf2_ros/include/tf2_ros/static_transform_broadcaster_node.hpp) & https://github.com/ros2/geometry2/blob/rolling/tf2_ros/include/tf2_ros/static_transform_broadcaster_visibility_control.h.

**Comment by mvukov** [comment]
btw, please merge the main branch.

**Comment by dbruggner** [comment]
@mvukov Thanks for the comments so far, do you have any other remarks?

**Comment by mvukov** [comment]
OK, I fixed the CI on the main branch. Please fix the failing test and then the PR is good to be merged.

**Comment by mvukov** [comment]
BTW, when you fix the yaml file, please regenerate the _impl.bzl file as well. Thanks in advance.

**Comment by dbruggner** [comment]
Done

**Review by mvukov — APPROVED** [review]
Very nice! LGTM! :rocket:
